CRM/Remoteevent/Localisation.php

```diff
@@ -79,31 +79,67 @@ public function localise($string, $context = [])
     /**
      * Localise a given string with this localisation.
      *
-     * @param string $string
+     * @param string $text
      *   The (English) string to localise.
      *
-     * @param array $context
+     * @param array $params
      *   Localisation parameters or variables.
      *
      * @return string
      *   The localised version of the string.
      *
      * @see \ts()
      */
-    public function ts($string, $context = []) {
-        if (empty($this->locale)) {
-            // No changes, used for pot extraction.
-            return $string;
+    public function ts($text, $params = [])
+    {
+        if (!isset($this->locale)) {
+          // No changes, used for pot extraction.
+          return $text;
+        }
+
+        static $bootstrapReady = FALSE;
+        static $lastLocale = NULL;
+        static $i18n = NULL;
+        static $function = NULL;
+
+        // TODO: Actually, this is obsolete, as custom translation functions
+        //       don't take the locale as an argument.
+        // When the settings become available, lookup customTranslateFunction.
+        if (!$bootstrapReady) {
+            $bootstrapReady = (bool) \Civi\Core\Container::isContainerBooted();
```

Comment:
Is this code actually called without the container being booted?

Reply:
I don't think so. Anyway it is only relevant if there's a custom translation function, which I haven't seen implemented anywhere yet. Also, all of this is just copied from Core's `ts()`.